dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/jaxrs/model/AbstractResourceInfo.java

```diff
@@ -89,6 +89,87 @@
+        CONTEXT_PROPS.put("org.apache.cxf.jaxrs.provider.StringTextProvider", ProviderContextInfo.NO_PROCESSING_REQUIRED);
+        CONTEXT_PROPS.put("org.apache.cxf.jaxrs.provider.PrimitiveTextProvider", ProviderContextInfo.NO_PROCESSING_REQUIRED);
+        CONTEXT_PROPS.put("org.apache.cxf.jaxrs.provider.JAXBElementProvider", new ProviderContextInfo(null, Collections.singleton("setMessageContext")));
+        CONTEXT_PROPS.put("org.apache.cxf.jaxrs.provider.AbstractJAXBProvider", ProviderContextInfo.NO_PROCESSING_REQUIRED);
+        CONTEXT_PROPS.put("org.apache.cxf.jaxrs.provider.JAXBElementTypedProvider", new ProviderContextInfo(null, Collections.singleton("setMessageContext")));
+        CONTEXT_PROPS.put("com.ibm.ws.jaxrs20.providers.multipart.IBMMultipartProvider", new ProviderContextInfo(Collections.singleton("mc"), null));
+        CONTEXT_PROPS.put("org.apache.cxf.jaxrs.provider.MultipartProvider", new ProviderContextInfo(Collections.singleton("mc"), null));
+        CONTEXT_PROPS.put("org.codehaus.jackson.jaxrs.JacksonJaxbJsonProviderWrapper", new ProviderContextInfo(null, null));
+        CONTEXT_PROPS.put("org.codehaus.jackson.jaxrs.JacksonJaxbJsonProvider", new ProviderContextInfo(null, null));
+        CONTEXT_PROPS.put("org.codehaus.jackson.jaxrs.JacksonJsonProvider", new ProviderContextInfo(Collections.singleton("_providers"), null));
+        CONTEXT_PROPS.put("org.codehaus.jackson.Versioned", ProviderContextInfo.NO_PROCESSING_REQUIRED);
+        CONTEXT_PROPS.put("com.ibm.ws.jaxrs20.providers.jsonp.JsonPProvider", ProviderContextInfo.NO_PROCESSING_REQUIRED);
+        CONTEXT_PROPS.put("org.apache.cxf.jaxrs.model.wadl.WadlGenerator", ProviderContextInfo.NO_PROCESSING_REQUIRED);
+        CONTEXT_PROPS.put("com.ibm.ws.jaxrs20.providers.customexceptionmapper.CustomWebApplicationExceptionMapper", ProviderContextInfo.NO_PROCESSING_REQUIRED);
+        CONTEXT_PROPS.put("com.ibm.ws.jaxrs20.security.LibertyAuthFilter", ProviderContextInfo.NO_PROCESSING_REQUIRED);
+        CONTEXT_PROPS.put("org.eclipse.microprofile.rest.client.ext.ResponseExceptionMapper", ProviderContextInfo.NO_PROCESSING_REQUIRED);
+        CONTEXT_PROPS.put("org.eclipse.microprofile.rest.client.ext.AsyncInvocationInterceptorFactory", ProviderContextInfo.NO_PROCESSING_REQUIRED);
+    }
+
+    private static class ProviderContextInfo {
+        static final ProviderContextInfo NO_PROCESSING_REQUIRED = new ProviderContextInfo();
+
+        // Indicates that this class AND its super classes and/or interface(s) do not have a @Context annotations
+        // on fields or methods.
+        final boolean processingRequired;
+
+        // The field names in the associated class that have a @Context annotation.  Null indicates none.
+        final Set<String> fieldNames;
+
+        // The method names in the associated class that have a @Context annotation.  Null indicates none.
+        final Set<String> methodNames;
+
+        private ProviderContextInfo() {
```

Comment:
We could probably avoid this constructor entirely and have the `NO_PROCESSING_REQUIRED` constant just pass nulls in to the other constructor (and set `processingRequired = fields !=null && methods != null` in that other constructor).

Reply:
Originally I had that, and had to pass in a boolean to say whether processing was required or not because you may not have fields or methods in the current Class that have @Context annotations, but a parent class may.  In that case you still need to process the class and do its parent / grandparent classes.  I am leaving it as it is.  The way it is coded is to avoid having multiple people new up the no processing required version and create additional memory that is not necessary.